ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -349,7 +350,10 @@ translateUTxOByronToShelley (CC.UTxO utxoByron) =
     -- bytes. We don't care about the type that is hashed, which will differ
     -- going from Byron to Shelley, we just use the hashes as IDs.
     translateTxId :: CC.TxId -> SL.TxId sc
-    translateTxId = SL.TxId . Hash.UnsafeHash . Hashing.hashToBytes
+    translateTxId =
+      SL.TxId . fromMaybe (error "Called hashFromBytes with input of wrong length")
```

Comment:
Introduce a helper that does the `fromMaybe`, because you need it in tons of places.